src/engine/GraphStoreProtocol.h

```diff
@@ -0,0 +1,134 @@
+        break;
+      }
+      default: {
+        // Unsupported media type -> 415 Unsupported Media Type
+        throw std::runtime_error(absl::StrCat(
+            "Mediatype \"", ad_utility::toString(contentType.value()),
+            "\" is not supported for SPARQL Graph Store HTTP "
+            "Protocol in QLever."));
+      }
+    }
+    ParsedQuery res;
+    auto transformTurtleTriple = [&graph](const TurtleTriple& triple) {
+      AD_CORRECTNESS_CHECK(triple.graphIri_.isId() &&
+                           triple.graphIri_.getId() ==
+                               qlever::specialIds().at(DEFAULT_GRAPH_IRI));
+      SparqlTripleSimpleWithGraph::Graph g{std::monostate{}};
+      if (std::holds_alternative<GraphRef>(graph)) {
+        g = Iri(std::get<GraphRef>(graph).toStringRepresentation());
+      }
+      return SparqlTripleSimpleWithGraph(triple.subject_, triple.predicate_,
+                                         triple.object_, g);
+    };
+    updateClause::GraphUpdate up{
+        ad_utility::transform(triples, transformTurtleTriple), {}};
+    res._clause = parsedQuery::UpdateClause{up};
+    return res;
+  }
+  FRIEND_TEST(GraphStoreProtocolTest, transformPost);
+
+  static ParsedQuery transformGet(const GraphOrDefault& graph) {
+    ParsedQuery res;
+    res._clause = parsedQuery::ConstructClause(
+        {{Variable("?s"), Variable("?p"), Variable("?o")}});
+    res._rootGraphPattern = {};
+    parsedQuery::GraphPattern selectSPO;
+    selectSPO._graphPatterns.emplace_back(parsedQuery::BasicGraphPattern{
+        {SparqlTriple(Variable("?s"), "?p", Variable("?o"))}});
+    if (std::holds_alternative<ad_utility::triple_component::Iri>(graph)) {
+      parsedQuery::GroupGraphPattern selectSPOWithGraph{
+          std::move(selectSPO),
+          std::get<ad_utility::triple_component::Iri>(graph)};
+      res._rootGraphPattern._graphPatterns.emplace_back(
+          std::move(selectSPOWithGraph));
+    } else {
+      AD_CORRECTNESS_CHECK(std::holds_alternative<DEFAULT>(graph));
+      res._rootGraphPattern = std::move(selectSPO);
+    }
+    return res;
+  }
+  FRIEND_TEST(GraphStoreProtocolTest, transformGet);
+
+ public:
+  // Every Graph Store Protocol requests has equivalent SPARQL Query or Update.
+  // Transform the Graph Store Protocol request into it's equivalent Query or
+  // Update.
+  static ParsedQuery transformGraphStoreProtocol(
+      const ad_utility::httpUtils::HttpRequest auto& rawRequest) {
+    ad_utility::url_parser::ParsedUrl parsedUrl =
+        ad_utility::url_parser::parseRequestTarget(rawRequest.target());
+    GraphOrDefault graph = extractTargetGraph(parsedUrl.parameters_);
+
+    using enum boost::beast::http::verb;
+    auto method = rawRequest.method();
+    if (method == get) {
+      return transformGet(graph);
+    } else if (method == put) {
+      throw std::runtime_error(
+          "PUT in the SPARQL Graph Store HTTP Protocol is not yet implemented "
+          "in QLever.");
```

Comment:
Can get rid of the code dupliccation (everything except for the method name is redundant, implement a lambda that does the throwing).